lib/types/webp.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var fs      = require('fs');
+var Promise = require('bluebird');
+var pread   = Promise.promisify(fs.read);
+var utils   = require('../utils');
+
+function isWEBP (buffer) {
+  return 'RIFF' === utils.ascii(buffer, 0, 4) && 'WEBP' === utils.ascii(buffer, 8, 12) && 'VP8'  === utils.ascii(buffer, 12, 15);
+}
+
+function measure (path, fd) {
+  return pread(fd, new Buffer(100), 0, 100, 0)
+  .spread(function (bytesRead, buffer) {
+    var chunkHeader = buffer.toString('ascii', 12, 16);
+    buffer = buffer.slice(20, 30);
+
+    if (chunkHeader === 'VP8 ' && buffer[0] !== 0x2f) {
+      return {
+        type: 'webp',
+        pages: [{
+          width: buffer.readInt16LE(6) & 0x3fff,
+          height: buffer.readInt16LE(8) & 0x3fff
+        }]
+      };
+    }
+
+    var signature = buffer.toString('hex', 3, 6);
+    if (chunkHeader === 'VP8L' && signature !== '9d012a') {
+      return {
+        type: 'webp',
+        pages: [{
+          width: 1 + (((buffer[2] & 0x3F) << 8) | buffer[1]),
+          height: 1 + (((buffer[4] & 0xF) << 10) | (buffer[3] << 2) | ((buffer[2] & 0xC0) >> 6))
+        }]
+      };
+    }
+
+    return Promise.reject(new Error('Invalid WEBP file'));
```

Comment:
For other files I just copied a valid image and manually edited the bytes to make it corrupt. Seems like that would work in this case too.